### ISSUE TITLE
PT4AL: Using Self-Supervised Pretext Tasks for Active Learning

### ISSUE THREAD

**Issue by zhuoxiao-chen** [zhuoxiao-chen/please_read_paper_every_day#10]
### Paper Link
https://www.ecva.net/papers/eccv_2022/papers_ECCV/papers/136860583.pdf
## Motivation

<img width="700" alt="image" src="https://user-images.githubusercontent.com/50613939/203516136-6c05132d-5454-4bd5-9a3b-191c83c560ef.png">

- There is a high correlation between self-supervised pretext tasks and downstream tasks, and thus pretext tasks can be utilized for active learning.
- Spearman’s rank correlation denoted as ρ is calculated on the full test set.

## Methodology

<img width="700" alt="image" src="https://user-images.githubusercontent.com/50613939/203518083-d8c98894-66f6-4d45-ab54-b67da68b55fb.png">

- **Diversity**: unlabeled data are sorted in descending order by their pretext task loss, and split into batches to be used for each AL iteration.
- **Uncertainty**: Starting from the batch containing data with the highest losses, the most uncertain K data points are sampled from each batch, based on the posterior class probability of the previous main task learner.

### Pretext Task Learning for Batch Split

$$
\operatorname{loss}\left(x_i, \theta_p\right)=\frac{1}{k} \sum_{y=1}^k \mathcal{L}_{C E}\left(F_p\left(g\left(x_i \mid y\right) \mid \theta_p\right), y\right)
$$

- Rotation prediction task ( $0^{\circ}, 90^{\circ}, 180^{\circ}, 270^{\circ}$ degrees).
- $g$ rotation operator.
- $F_p$ represents the probability distribution of the input image rotated by label $y$.
- Given the pretext task loss values of the unlabeled data, sort the losses in descending order, divided into $I$ batches of equal size, the number of $I$  is equal to the number of AL iterations, select $K$ samples from each batch at each AL iteration. 
- Unlabeled data are split into 10 Batches. 

### In-batch Sampling
<img width="600" alt="image" src="https://user-images.githubusercontent.com/50613939/203677463-aa6d9447-a043-4a65-bf3b-37187ed385e3.png">

### Batch Model  Active Learning (BMAL)
Most active learning approaches, however, have focused on selecting only one unlabeled instance at one time, while retraining the classifier on each iteration. When the training process is hard or time consuming, this repeated retraining is inefficient. Furthermore, if a parallel labeling system is available, a single instance selection system can make wasteful use of the resource. Thus, a **batch mode** active learning strategy that selects multiple instances each time is more appropriate under these circumstances. 

### Diversity and Uncertainty Feature Visualization 
<img width="700" alt="image" src="https://user-images.githubusercontent.com/50613939/203548347-62da5d1b-e252-4e3a-9b12-49c0f43ba567.png">

- The sampled points are either concentrated on the class boundaries or evenly located in the class distributions.

### Performance on Image Classification Task
<img width="700" alt="image" src="https://user-images.githubusercontent.com/50613939/203549385-4f37903e-bb5c-4f44-b247-7a48e1cc9db2.png">

### Performance on Semantic Segmentation Task
<img width="700" alt="image" src="https://user-images.githubusercontent.com/50613939/203548522-7e05d00c-5838-403f-9ad8-ad4068f00679.png">

- Recomposed the CIFAR10 dataset: airplane-500, automobile-1,000, bird-1,500, cat-2,000, deer-2,500, dog-3,000, frog-3,500, horse-4,000, ship-4,500 and truck-5,000.
- Data from classes with little training data generally have high loss.
- Since PT4AL samples from data batches with high to low loss, it can sample in a class-balanced way even in imbalanced settings.

### Ablation Study
<img width="700" alt="image" src="https://user-images.githubusercontent.com/50613939/203548770-8c13c31b-c6f9-4702-b2ca-15628272e987.png">

- High loss batch first method starts the first iteration with the batch containing the highest pretext task losses, then moves to batches with lower losses for consecutive iterations. 
- Low loss batch first is in reverse.
